ios/RNTPTDocumentView.m

```diff
@@ -4582,6 +4595,50 @@ - (NSString *) getDocumentPath {
     return self.currentDocumentViewController.coordinatedDocument.fileURL.path;
 }
 
+#pragma mark - Get All Fields
+
+- (NSMutableArray<NSDictionary *> *)getAllFieldsForDocumentViewTag:(int)pageNumber
+{
+    PTPDFViewCtrl *pdfViewCtrl = self.currentDocumentViewController.pdfViewCtrl;
+    if (!pdfViewCtrl) {
+        return nil;
+    }
+    PTPDFDoc * doc = [pdfViewCtrl GetDoc];
+    NSMutableArray<NSDictionary *> *resultMap = [[NSMutableArray alloc] init];
+    NSError *error;
+    if(doc){
+        PTPage *page = [doc GetPage:pageNumber];
```

Comment:
I think this needs a read lock, can you put the read lock from line 4621 earlier? i.e. make sure all the doc access calls are in the read lock.